### PR TITLE
add reusable submission workflow, submission workflows for each environment

### PR DESCRIPTION
**Pull request by nr-kkenney**
## Summary
This PR adds in a new reusable workflow which can validate or submit install plans & quickstarts for a given environment. Additionally, a new workflow has been added for each environment. This allows us to run changes for only a single environment as needed.

Finally, the core submission workflow has been updated to utilize this new reusable workflow. With this change, the overall process for when a pr is merged is now slightly different. Staging runs before the other environments, with production & eu-production running simultaneously. For each environment, install plans are first added, then quickstarts. 

Previously, the process was:
 1. staging install plans added.
 2. production install plans added.
 3. eu-production install plans added.
 4. staging quickstarts added.
 5. production quickstarts added.
 6. eu-production quickstarts added.

## Screenshots
Here is an example run summary with the new setup: 
Screen Shot 2022-04-06 at 4 54 21 PM

## Links
Resolves: #934

### CODE REVIEW

File: .github/workflows/pr-merged.yml [hunk 1]
```diff
@@ -133,97 +134,35 @@ jobs:
...
-          NR_API_TOKEN: ${{ secrets.NR_API_TOKEN_EU }}
-        run: |
-          URL="https://api.github.com/repos/${GITHUB_REPOSITORY}/pulls/${{ steps.get_pr_number.outputs.pr-number }}/files"
-          cd utils && yarn create-validate-pr-quickstarts $URL ${DRY_RUN}
+  staging:
+    needs: [generate-uuid, get-pr-number]
+    uses: ./.github/workflows/reusable.quickstart_submission.yml
+    with:
+      pr-number: ${{ needs.get-pr-number.outputs.pr-number }}
+      dry-run: true
```
Comment (nr-kkenney) [inline .github/workflows/pr-merged.yml:142]: these should be false for real submission 🙈

File: .github/workflows/quickstarts.eu-production.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      pr-number:
+        description: 'PR to run against'
+        required: true
+
+jobs:
+  eu-production:
+    uses: ./.github/workflows/reusable.quickstart_submission.yml
+    with:
+      pr-number: ${{ github.event.inputs.pr-number }}
+      dry-run: true
```
Comment (nr-kkenney) [inline .github/workflows/quickstarts.eu-production.yml:15]: same comment as above. should be false for real submissions.

**Review by aswanson-nr — COMMENTED** [review]
Clean!! this looks great! 
Can we use this reusable workflow for the validation as well?
Also I remember talking about having the manual publishing workflows take in the quickstart ID instead of working with a PR, but that would definitely require changes on the script side, so this looks fantastic for the first pass!

**Comment by nr-kkenney** [comment]
I'll also copy this comment over to the ticket.

With how i set things up, having the validation workflows use the reusable workflow would be a bit messy. The reusable workflow does both install plan & quickstart stuff in one run. The validation we currently have in the repo is separate for install plans & quickstarts, with no clean way to use the reusable workflow. Additionally, the validation workflows actually have more jobs than the reusable workflow does -- they are validating multiple things, sometimes using many scripts. The reusable workflow only validates against the API (schema validation).

i could setup two reusable workflows — one for install plans, one for quickstarts — but i actually think we dont want this, since it might be on the table to actually just have one entrypoint script in the future — the CLI.

**Review by aswanson-nr — APPROVED** [review]

**Pull request merged**